requirements.txt

```diff
@@ -56,6 +56,9 @@ pyopenssl==21.0.0 ; python_version < '3.12'
 pyopenssl==24.1.0 ; python_version >= '3.12' # (Noble) min 23.2.0, pinned for compatibility with cryptography==42.0.8 and security patches
 PyPDF2==1.26.0 ; python_version <= '3.10'
 PyPDF2==2.12.1 ; python_version > '3.10'
+pymupdf==1.20.2 ; python_version <= '3.10'
+pymupdf==1.21.1 ; python_version > '3.10' and python_version < '3.12'
+pymupdf==1.23.6 ; python_version >= '3.12'
```

Comment:
Is it really necessary ? Introducing new dependencies must always be check thoughtfully as these are new attack vector and complexion the installation. We rely on the version available in package distrubition of Debian or Ubuntu. I am not sure pymupdf is available in Debian. Therefore this dependency may not be available.
![image](https://github.com/user-attachments/assets/8309e565-f177-4df3-ba7b-95f7a576e46d)
If you don't rely on a lot of functionality of pymupdf (which is a huge library !), you may have to either recode the needed parts or rely on a smallest library available in Debian/ubuntu.
Thanks :-)